astropy/io/fits/tests/test_connect.py

```diff
@@ -57,13 +60,35 @@ def equal_data(a, b):
     return all(np.all(a[name] == b[name]) for name in a.dtype.names)
 
 
+@contextlib.contextmanager
+def working_directory(path):
+    """
+    A context manager changing the working directory to the given
+    path. The directory is changed back on exit.
+    """
+
+    cwd = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(cwd)
```

Comment:
Might be a more generally useful context manager, I'm surprised we don't already have one in `utils` ?

Reply:
Maybe rename to `change_dir`, `change_cwd` or something similar ?

Reply:
Good news: we can actually use the standard lib here ! https://docs.python.org/3/library/contextlib.html#contextlib.chdir

Reply:
Ah right it's in the stdlib now !